plugin/trino-bigquery/src/test/java/io/trino/plugin/bigquery/BaseBigQueryConnectorTest.java

```diff
@@ -1146,6 +1152,186 @@ public void testExecuteProcedureWithInvalidQuery()
         assertQueryFails("CALL system.execute('invalid')", ".*Syntax error: Unexpected identifier.*");
     }
 
+    @Test
+    public void testLimitPushdownWithExternalTable()
+    {
+        String externalTableName =  TEST_SCHEMA + ".region_external_table_" + randomNameSuffix();
+        try {
+            onBigQuery("CREATE EXTERNAL TABLE " + externalTableName + " OPTIONS (format = 'CSV', uris = ['gs://" + gcpStorageBucket + "/tpch/tiny/region.csv'])");
+            assertLimitPushdownOnRegionTable(getSession(), externalTableName);
+        }
+        finally {
+            onBigQuery("DROP EXTERNAL TABLE " + externalTableName);
+        }
+    }
+
+    @Test
+    public void testLimitPushdownWithView()
+    {
+        try (TestView testView = new TestView(bigQuerySqlExecutor::executeQuery, TEST_SCHEMA + ".region_view_", "SELECT * FROM tpch.region")) {
+            // view with materialization uses storage api, with storage api limit pushdown is not supported
+            assertThat(query("SELECT * FROM " + testView.getName() + " LIMIT 3"))
+                    .skipResultsCorrectnessCheckForPushdown()
+                    .isNotFullyPushedDown(node(LimitNode.class, anyTree(node(TableScanNode.class))));
+
+            Session sessionWithSkipViewMaterialization = Session.builder(getSession())
+                    .setCatalogSessionProperty("bigquery", "skip_view_materialization", "true")
+                    .build();
+            assertLimitPushdownOnRegionTable(sessionWithSkipViewMaterialization, testView.getName());
+        }
+    }
+
+    @Test
+    public void testLimitPushdownWithMaterialziedView()
+    {
+        String mvName =  TEST_SCHEMA + ".region_mv_" + randomNameSuffix();
+        try {
+            onBigQuery("CREATE MATERIALIZED VIEW " + mvName + " AS SELECT * FROM tpch.region");
+
+            // materialized view with materialization uses storage api, with storage api limit pushdown is not supported
+            assertThat(query("SELECT * FROM " + mvName + " LIMIT 3"))
+                    .skipResultsCorrectnessCheckForPushdown()
+                    .isNotFullyPushedDown(node(LimitNode.class, anyTree(node(TableScanNode.class))));
+
+            Session sessionWithSkipViewMaterialization = Session.builder(getSession())
+                    .setCatalogSessionProperty("bigquery", "skip_view_materialization", "true")
+                    .build();
+            assertLimitPushdownOnRegionTable(sessionWithSkipViewMaterialization, mvName);
+        }
+        finally {
+            onBigQuery("DROP MATERIALIZED VIEW " + mvName);
```

Comment:
This statement may fail if `CREATE MATERIALIZED VIEW` failed. I would move `CREATE MATERIALIZED VIEW` before `try`. Same for other tests.